src/platform/plugins/shared/dashboard/server/content_management/v3/transforms/out/control_group_out_transforms.ts

```diff
@@ -17,21 +19,46 @@ import {
   DEFAULT_IGNORE_PARENT_SETTINGS,
   type ParentIgnoreSettings,
 } from '@kbn/controls-plugin/common';
-import type { DashboardSavedObjectAttributes } from '../../../../dashboard_saved_object';
+
 import type { ControlGroupAttributes } from '../../types';
 import { transformControlsState } from './control_state_out_transforms';
 
-export const transformControlGroupOut: (
-  controlGroupInput: NonNullable<DashboardSavedObjectAttributes['controlGroupInput']>
-) => ControlGroupAttributes = flow(
-  transformControlGroupSetDefaults,
-  transformControlGroupProperties
-);
+/**
+ * Transform an unknown `controlGroupInput` from the dashboard SO to a valid `controlGroupInput` for the CM schema
+ */
+const controlGroupSavedObjectSchema = schema.object({
+  panelsJSON: schema.string(),
+  controlStyle: schema.maybe(schema.string()),
+  chainingSystem: schema.maybe(schema.string()),
+  ignoreParentSettingsJSON: schema.maybe(schema.string()),
+  showApplySelections: schema.maybe(schema.boolean()),
+});
+type ControlGroupSavedObject = TypeOf<typeof controlGroupSavedObjectSchema>;
+
+export const isControlGroupSavedObject = (
+  controlGroupInput: unknown
+): controlGroupInput is ControlGroupSavedObject => {
+  try {
+    return Boolean(controlGroupSavedObjectSchema.validate(controlGroupInput));
```

Comment:
This is clever. But I wonder, don't we really only need to check if all _required_ attributes are in the `controlGroupInput`, e.g. `panelsJSON` to validate this?